tests/test_E2_economics.py

```diff
@@ -86,50 +123,115 @@
 
 def test_all_cost_info_parameters_added_to_dict_asset():
     """Tests whether the function get_costs is adding all the calculated costs to dict_asset."""
+    print(dict_economic)
     E2.get_costs(dict_asset, dict_economic)
+
+    # Note: The valid calculation of the costs is tested with test_benchmark_KPI.py, Test_Economic_KPI.test_benchmark_Economic_KPI_C2_E2()
     for k in (
         COST_DISPATCH,
-        COST_OM_FIX,
+        COST_OM,
         COST_TOTAL,
-        COST_OM_TOTAL,
+        COST_OPERATIONAL_TOTAL,
         ANNUITY_TOTAL,
         ANNUITY_OM,
     ):
         assert k in dict_asset
 
 
-def test_add_costs_and_total():
-    """Tests if new costs are adding to current costs correctly and if dict_asset is being updated accordingly."""
-    current_costs = 10000
-    new_cost = 5000
-    total_costs = E2.add_costs_and_total(
-        dict_asset, "new_cost", new_cost, current_costs
+test_all_cost_info_parameters_added_to_dict_asset()
+
+
+def test_calculate_costs_replacement():
+    cost_replacement = E2.calculate_costs_replacement(
+        specific_replacement_of_initial_capacity=5,
+        specific_replacement_of_optimized_capacity=10,
+        initial_capacity=1,
+        optimized_capacity=10,
+    )
+    assert cost_replacement == 5 * 1 + 10 * 10
+
+
+def test_calculate_operation_and_management_expenditures():
+    operation_and_management_expenditures = E2.calculate_operation_and_management_expenditures(
+        specific_om_cost=5, installed_capacity=10, optimized_add_capacity=10
+    )
+    assert operation_and_management_expenditures == 100
+
+
+def test_calculate_total_asset_costs_over_lifetime():
+    total = E2.calculate_total_asset_costs_over_lifetime(
+        costs_investment=300, cost_operational_expenditures=200
+    )
+    assert total == 500
+
+
+def test_calculate_costs_upfront_investment():
+    costs = E2.calculate_costs_upfront_investment(
+        specific_cost=100, capacity=5, development_costs=200
     )
-    assert total_costs == new_cost + current_costs
-    assert "new_cost" in dict_asset
+    assert costs == 700
+
+
+def test_calculate_total_capital_costs():
+    total_capital_expenditure = E2.calculate_total_capital_costs(
+        upfront=300, replacement=100
+    )
+    assert total_capital_expenditure == 400
+
+
+def test_calculate_total_operational_expenditures():
+    total_operational_expenditures = E2.calculate_total_operational_expenditures(
+        operation_and_management_expenditures=100, dispatch_expenditures=500
+    )
+    assert total_operational_expenditures == 600
+
+
+asset = "an_asset"
```

Comment:
better to make them global variables for this module at the top of the model

Reply:
They are only needed in the few tests below, so I wanted them close to those tests.